src/types/utils.ts

```diff
@@ -269,3 +262,179 @@ export function typeNameToTypeNode(astT: TypeName): TypeNode {
 
     throw new Error(`NYI converting AST Type ${astT.print()} to SType`);
 }
+
+/**
+ * Computes a `TypeNode` equivalent of given `astT`,
+ * specialized for location `loc` (if applicable).
+ */
+export function typeNameToSpecializedTypeNode(astT: TypeName, loc: DataLocation): TypeNode {
+    return specializeType(typeNameToTypeNode(astT), loc);
+}
+
+/**
+ * Given a `VariableDeclaration` node `decl` compute the `TypeNode` that corresponds to the variable.
+ * This takes into account the storage location of the `decl`.
+ */
+export function variableDeclarationToTypeNode(decl: VariableDeclaration): TypeNode {
+    assert(decl.vType !== undefined, "Expected {0} to have type", decl);
+
+    return typeNameToSpecializedTypeNode(decl.vType, decl.storageLocation);
+}
+
+export function enumToIntType(decl: EnumDefinition): IntType {
+    const length = decl.children.length;
+
+    let size: number | undefined;
+
+    for (let n = 8; n <= 32; n += 8) {
+        if (length < 2 ** n) {
+            size = n;
+
+            break;
+        }
+    }
+
+    assert(
+        size !== undefined,
+        "Unable to detect enum type size - member count exceeds 2 ** 32",
+        decl
+    );
+
+    return new IntType(size, false);
+}
+
+/**
+ * Converts `type` to value type. Following rules are applied:
+ * 1. Contract definitions turned to address.
+ * 2. Enum definitions turned to uint of minimal fitting size.
+ */
+function toValueType(type: TypeNode, encoderVersion: ABIEncoderVersion): TypeNode {
```

Comment:
The name of the function is a bit general. Maybe something like `toAccessorType`? Not sure. Also not a blocker, might merge as is.

Reply:
Irrelevant since bf45d33